### PR TITLE
api: introduce ElementHandle.waitForSelector

### PR DESCRIPTION
**Pull request by dgozman**
This is similar to Frame.waitForSelector, but relative to the handle.

References #3366.

### CODE REVIEW

File: src/dom.ts [hunk 1]
```diff
@@ -636,6 +636,33 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
...
   }
 
+  async waitForSelector(selector: string, options: types.WaitForElementOptions = {}): Promise<ElementHandle<Element> | null> {
+    const { state = 'visible' } = options;
+    if (!['attached', 'detached', 'visible', 'hidden'].includes(state))
+      throw new Error(`state: expected one of (attached|detached|visible|hidden)`);
+    const info = selectors._parseSelector(selector);
+    const task = waitForSelectorTask(info, state, this);
+    return this._page._runAbortableTask(async progress => {
+      progress.logger.info(`waiting for selector "${selector}"${state === 'attached' ? '' : ' to be ' + state}`);
```
Comment (pavelfeldman) [inline src/dom.ts:646]: Is there a way to extract anything?

**Review by pavelfeldman — APPROVED** [review]

File: src/dom.ts (hunk 1 above)
Comment (dgozman) [reply src/dom.ts:646]: I can extract the finishing part.

**Comment by creage** [comment]
Thank you mates so much for implementing this method! We need this feature to be released ASAP, please! :)

**Comment by mxschmitt** [comment]
> Thank you mates so much for implementing this method! We need this feature to be released ASAP, please! :)

If you want to use the latest developer edition, feel free to install it by `npm install playwright@next`. There you can use this already.

**Pull request merged**